File: weather_script.js
const fetch = require('node-fetch');

async function obtenerClima(lat, lon, apiKey) {
    const url = `https://api.openweathermap.org/data/3.0/onecall?lat=${lat}&lon=${lon}&exclude=minutely,hourly,daily,alerts&appid=${apiKey}&units=metric`;
    try {
        const response = await fetch(url);
        if (response.ok) {
            const data = await response.json();
            return data;
        } else {
            console.error('Error al obtener los datos del clima:', response.statusText);
            return null;
        }
    } catch (error) {
        console.error('Error al realizar la solicitud:', error);
        return null;
    }
}

function mostrarClima(clima) {
    if (clima) {
        const tempActual = clima.current.temp;
        const descripcion = clima.current.weather[0].description;
        console.log(`La temperatura actual es ${tempActual}°C con ${descripcion}.`);
    } else {
        console.log('No se pudo obtener el clima.');
    }
}

async function recuperarDatosClima() {
    const lat = parseFloat(prompt('Introduce la latitud: '));
    const lon = parseFloat(prompt('Introduce la longitud: '));
    const apiKey = prompt('Introduce tu API key de OpenWeatherMap: ');

    const clima = await obtenerClima(lat, lon, apiKey);
    mostrarClima(clima);
}

recuperarDatosClima();

File: weather_script.py
import requests

def obtener_clima(lat, lon, api_key):
    url = f"https://api.openweathermap.org/data/3.0/onecall?lat={lat}&lon={lon}&exclude=minutely,hourly,daily,alerts&appid={api_key}&units=metric"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json()
    else:
        return None

def mostrar_clima(clima):
    if clima:
        temp_actual = clima['current']['temp']
        descripcion = clima['current']['weather'][0]['description']
        print(f"La temperatura actual es {temp_actual}°C con {descripcion}.")
    else:
        print("No se pudo obtener el clima.")

def recuperar_datos_clima():
   
    lat = float(input("Introduce la latitud: "))
    lon = float(input("Introduce la longitud: "))
    api_key = input("Introduce tu API key de OpenWeatherMap: ")
    
    clima = obtener_clima(lat, lon, api_key)
    mostrar_clima(clima)

if __name__ == "__main__":
    recuperar_datos_clima()